src/Concerns/WithClientAutoInitializeTrait.php

```diff
@@ -2,16 +2,13 @@
 
 namespace Liior\SymfonyTestHelpers\Concerns;
 
-use Liior\SymfonyTestHelpers\Exception\ClientNotCreatedException;
-use Symfony\Bundle\FrameworkBundle\KernelBrowser;
-use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
-use Symfony\Component\BrowserKit\AbstractBrowser;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
 trait WithClientAutoInitializeTrait
 {
     protected function setUp(): void
     {
-        if (\method_exists($this, 'initializeClient')) {
+        if ($this instanceof WithClientTrait) {
```

Comment:
This change brings a new bug. 

`dump($this instanceof WithClientTrait)` will give `false` so it never calls the `initializeClient()` method. I saw that you could use `class_uses()` to get an array of traits used by the class, but it only works with traits that are explicitly declared in the class.

For instance, in a class that extends `Liior\SymfonyTestHelpers\WebTestCase`, it would give you an empty array because Traits are not declared in the class itself but in the parent.

So for now, the best solution would stay the good old `method_exists($this, 'initializeClient')`

Reply:
An other comment for the use of `AbstractBrowser `as type for `$this->client` : `AbstractBrowser` does not have a `getContainer()` method, but it is often used to gain access to client services (as I understood it). The `KernelBrowser `has this method. Maybe it would be usefull for DX to keep `KernelBrowser` as a type, even if it may break DIP ?